### ISSUE TITLE
Meal plan with notes throws error

### ISSUE THREAD

**Issue by btoconnor** [zanix/MMM-MealieMenu#2]
Occasionally we make meal plans where some days just use the "Note" feature, rather than linking to a recipe. I created one on our current meal plan, and the `MMM-MealieMenu` module threw an error:

```
MMM-MealieMenu.js:279 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'name')
    at Class.formatMeals (MMM-MealieMenu.js:279:27)
    at Class.socketNotificationReceived (MMM-MealieMenu.js:172:49)
    at module.js:235:9
    at Socket.<anonymous> (socketclient.js:38:4)
    at Emitter.emit (index.mjs:136:20)
    at Socket.emitEvent (socket.js:498:20)
    at Socket.onevent (socket.js:485:18)
    at MMSocket.socket.onevent (socketclient.js:32:11)
    at Socket.onpacket (socket.js:455:22)
    at Emitter.emit (index.mjs:136:20)
```

It looks like this function is assuming all of the entries will be linked recipes.

These can be added in mealie by toggling the "Note Only" flag:
<img width="509" alt="image" src="https://github.com/zanix/MMM-MealieMenu/assets/81829/26dd79ec-9a54-4d74-b1bb-8db3aaf55f81">

**Comment by btoconnor** [comment]
It looks like this can be fixed by checking if `meal.recipe` exists before pulling in the name and the photo.

This diff makes it so there is no longer an error:
```diff
diff --git a/MMM-MealieMenu.js b/MMM-MealieMenu.js
index 2a5fac5..cc82c71 100644
--- a/MMM-MealieMenu.js
+++ b/MMM-MealieMenu.js
@@ -276,11 +276,11 @@ Module.register("MMM-MealieMenu", {
     const formatted = [];
     for (const meal of meals) {
       formatted.push({
-        name: meal.recipe.name,
+        name: meal.recipe ? meal.recipe.name : meal.title,
         rawDate: meal.date,
         date: moment(meal.date).format(this.config.dateFormat),
         meal: this.typeToMealDisplay(meal.entryType),
-        photoUrl: `${this.config.host}/api/media/recipes/${meal.recipe.id}/images/min-original.webp`,
+        photoUrl: meal.recipe ? `${this.config.host}/api/media/recipes/${meal.recipe.id}/images/min-original.webp` : '',
         isToday: today.isSame(meal.date),
         shouldFade: this.config.fadePriorEntries && today.isAfter(meal.date)
       });
```
However, I think it might be nice to have a default image in case the recipe or the note doesn't have a picture.  I can open up a PR for this if you'd like, but wouldn't know what to do for the image.

**Comment by zanix** [comment]
> It looks like this function is assuming all of the entries will be linked recipes.

Yep, that is true. I have not used the "Note only" option in quite some time and I forgot it exists.

> However, I think it might be nice to have a default image in case the recipe or the note doesn't have a picture. I can open up a PR for this if you'd like, but wouldn't know what to do for the image.

Maybe using the Mealie icon would work? I could also add a `defaultImage` option so it can be overridden. I also need to check if the `meal.recipe.image` field is blank when the recipe doesn't have an image. I would want touse a default image in those cases too.

**Comment by btoconnor** [comment]
The Mealie icon makes sense to me - I only got a brief chance to look but it seemed like it was an SVG, at least on the meal plan page.  Not sure if that's true and / or how easy that would be to pull into this project.

I took a quick look and tested the module with a recipe that doesn't have an image, and the module here with my hotfix shows a broken image (not too surprising):
![image](https://github.com/zanix/MMM-MealieMenu/assets/81829/b41b8f4e-6417-4d1b-8f39-af616a9e7232)

BTW - thanks for releasing this module. I am in the process of migrating to MagicMirror, specifically so that I could get access to custom data for the Mealie meal plan.  I thought I'd have to make the module myself, but this does exactly what I wanted.

**Comment by zanix** [comment]
> The Mealie icon makes sense to me - I only got a brief chance to look but it seemed like it was an SVG, at least on the meal plan page. Not sure if that's true and / or how easy that would be to pull into this project.

I'm planning on including a PNG version with an orange background and white icon, which is how it looks in the header of Mealie.

![mealie](https://github.com/zanix/MMM-MealieMenu/assets/1580378/20632efb-5e63-4739-ac74-b91bc0059c16)

I will also add the `defaultImage` configuration option so it can be changed to anything else.

> BTW - thanks for releasing this module. I am in the process of migrating to MagicMirror, specifically so that I could get access to custom data for the Mealie meal plan. I thought I'd have to make the module myself, but this does exactly what I wanted.

Thanks! I previously used Paprika and the MMM-PaprikaMenu module, but then I switched to Mealie and needed this module.

I should be able to update the code later today.
If you have any other suggestions for improving this module, let me know.

**Comment by zanix** [comment]
The meal plan notes issue is fixed in fdfe2497d1643dabf4873ed273743ae27b81a8d0

I'll get the defaultImage setting added tomorrow.

**Comment by btoconnor** [comment]
Fix works for me, thanks!
